src/core/lib/matchers/matchers.cc

```diff
@@ -291,7 +291,7 @@ bool HeaderMatcher::Match(
     match = value.has_value() == present_match_;
   } else if (!value.has_value()) {
     // All other types fail to match if field is not present.
-    match = false;
+    return false;
```

Comment:
This block would be triggered for types that are not `Present match` and value is absent.
Envoy also returns failed match in that case, irrespective of the invert value.
https://github.com/envoyproxy/envoy/blob/4c76813087c07171996d2cd8a9c591604b9124d8/source/common/http/header_utility.cc#L142
@markdroth @ejona86 Please correct me if I am wrong. Thank you.

Reply:
Sorry, you're right -- I misread the code change you made.  I somehow thought you were changing the behavior of present_match to always return false, regardless of the value of invert_match.

Sorry for the confusion!